File: 백준/Silver/9536. 여우는 어떻게 울지？/여우는 어떻게 울지？.py
for i in range(int(input())):
    lst = list(input().split())
    words,ans = [],[]
    while 1:
        word = list(input().split())
        if " ".join(word) == "what does the fox say?":
            break
        words += [word[i] for i in range(len(word)) if i%2==0]
    for j in lst:
        if j not in words:
            ans.append(j)
    print(" ".join(ans))    

File: 백준/Silver/9536. 여우는 어떻게 울지？/여우는 어떻게 울지？.java
import java.io.*;
import java.util.*;

class Main {
    static int n, m;
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    public static void main(String[] args) throws IOException {
        n = Integer.parseInt(br.readLine());
        for (int j = 0; j < n; j++) {
            List<String> ans = new ArrayList<>();
            List<String> words = Arrays.asList(br.readLine().split(" "));
            while(true){
                st = new StringTokenizer(br.readLine());
                int count = st.countTokens();
                if(count == 5){
                    break;
                }

                for (int i = 0; i < 2; i++) {
                    st.nextToken();
                }
                String sound = st.nextToken();
                for (int i = 0; i < words.size(); i++) {
                    if(words.get(i).equals(sound)){
                        words.set(i," ");
                    }
                }
            }
            for (String word : words) {
                if(!word.trim().isEmpty()){
                    ans.add(word);
                }
            }

            System.out.println(String.join(" ",ans));
        }
    }
}